pkg/scheduler/plugins/elasticquota/core/group_quota_manager.go

```diff
@@ -0,0 +1,616 @@
+	}
+
+	// store Information of SystemQuotaGroup and DefaultQuotaGroup
+	defaultInfo := gqm.quotaInfoMap[extension.DefaultQuotaName].DeepCopy()
+	systemInfo := gqm.quotaInfoMap[extension.SystemQuotaName].DeepCopy()
+
+	// clear old structure
+	gqm.quotaInfoMap = make(map[string]*QuotaInfo)
+	gqm.quotaTreeWrapperMap = make(map[string]*QuotaTreeWrapper)
+	gqm.quotaTopoNode = make(map[string]*QuotaTopoNode)
+
+	// reset structure
+	gqm.quotaInfoMap[extension.DefaultQuotaName] = defaultInfo
+	gqm.quotaInfoMap[extension.SystemQuotaName] = systemInfo
+	gqm.quotaTreeWrapperMap[extension.RootQuotaName] = NewQuotaTreeWrapper(extension.RootQuotaName)
+	gqm.quotaTreeWrapperMap[extension.RootQuotaName].SetClusterTotalResource(gqm.totalResourceIndeed)
+	rootNode := newAllQuotaTopoNodes[extension.RootQuotaName]
+	gqm.resetAllGroupQuotaRecursiveNoLock(rootNode, newMaxQuotaMap, newMinQuotaMap, newSharedWeightMap)
+
+	// subGroup's topo relation may change; refresh the request/used from bottom to up
+	for quotaName, topoNode := range newAllQuotaTopoNodes {
+		gqm.quotaTopoNode[quotaName] = topoNode
+		if quotaName != extension.RootQuotaName && !topoNode.quotaInfo.IsParent {
+			gqm.updateGroupDeltaRequestNoLock(quotaName, childRequestMap[quotaName])
+			gqm.updateGroupDeltaUsedNoLock(quotaName, childUsedMap[quotaName])
+		}
+	}
+	for quotaName, topoNode := range newAllQuotaTopoNodes {
+		if quotaName != extension.RootQuotaName && quotaName != extension.SystemQuotaName && quotaName != extension.DefaultQuotaName &&
+			!topoNode.quotaInfo.IsParent {
+			gqm.refreshRuntimeNoLock(quotaName)
+		}
+	}
+
+	gqm.updateResourceKeyNoLock()
+}
+
+// ResetAllGroupQuotaRecursiveNoLock no need to lock gqm.lock
+func (gqm *GroupQuotaManager) resetAllGroupQuotaRecursiveNoLock(rootNode *QuotaTopoNode,
+	newMaxQuotaMap, newMinQuotaMap, newSharedWeightMap quotaResMapType) {
+	childGroupQuotaInfos := rootNode.GetChildGroupQuotaInfos()
+	for subName, topoNode := range childGroupQuotaInfos {
+		gqm.quotaTreeWrapperMap[subName] = NewQuotaTreeWrapper(subName)
+		// equal, not deepCopy
+		gqm.quotaInfoMap[subName] = topoNode.quotaInfo
+
+		gqm.doUpdateOneGroupMaxQuotaNoLock(topoNode.quotaInfo, newMaxQuotaMap[subName])
+		gqm.updateMinQuotaNoLock(topoNode.quotaInfo, newMinQuotaMap[subName])
+		gqm.doUpdateOneGroupSharedWeightNoLock(topoNode.quotaInfo, newSharedWeightMap[subName])
+
+		gqm.resetAllGroupQuotaRecursiveNoLock(topoNode, newMaxQuotaMap, newMinQuotaMap, newSharedWeightMap)
+	}
+}
+
+//doUpdateOneGroupMaxQuotaNoLock no need to lock gqm.lock
+func (gqm *GroupQuotaManager) doUpdateOneGroupMaxQuotaNoLock(quotaInfo *QuotaInfo, newMax v1.ResourceList) {
+	curToAllParInfos := gqm.getCurToAllParentGroupQuotaInfoNoLock(quotaInfo.Name)
+	quotaInfoLen := len(curToAllParInfos)
+	if quotaInfoLen <= 0 {
+		return
+	}
+
+	defer gqm.scopedLockForQuotaInfo(curToAllParInfos)()
+	if quotav1.Equals(quotaInfo.CalculateInfo.Max, newMax) {
+		return
+	}
+
+	if quotaInfoLen == 1 {
```

Comment:
these if\else if useless for you reset, because you udpate all max, then update all request, which means when you update max, the request is zero